code/game/objects/items/explosives/mine.dm

```diff
@@ -170,4 +178,20 @@
 		return
 
 	if(linked_claymore && ismob(A))
-		linked_claymore.Bumped(A)
\ No newline at end of file
+		linked_claymore.Bumped(A)
+
+/obj/item/explosive/mine/proc/mine_alert(mob/M)
+	if(!M)
+		return
+	var/notice = "<b>ALERT! [src] detonated. Hostile/unknown: [M] Detected at: [get_area(M)]. Coordinates: (X: [M.x], Y: [M.y]).</b>"
+	playsound(loc, 'sound/machines/warning-buzzer.ogg', 50, FALSE)
+	var/mob/living/silicon/ai/AI = new/mob/living/silicon/ai(src, null, null, 1)
+	AI.SetName("Smartmine Alert System")
+	AI.aiRadio.talk_into(AI,"[notice]","Theseus","announces")
+	qdel(AI)
+
+	switch(trigger_type) //Makes sure we announce first before detonation.
```

Comment:
Does this require a `switch`, if there's a single option?
Could be simplified with `if(trigger_type == "explosive" && tripwire)`